procrustes/psdp.py

```diff
@@ -23,18 +23,249 @@
 """Positive semidefinite Procrustes Module."""
 
 from math import inf, sqrt
-from typing import Optional
+from typing import Dict, Optional
 
 import numpy as np
 from numpy.linalg import multi_dot
 from procrustes.utils import compute_error, ProcrustesResult, setup_input_arrays
 import scipy
 from scipy.optimize import minimize
 
-__all__ = [
-    "psdp_woodgate",
-    "psdp_peng"
-]
+__all__ = ["psdp_woodgate", "psdp_peng", "psdp_opt"]
+
+
+def psdp_opt(
+    a: np.ndarray,
+    b: np.ndarray,
+    options: Dict,
```

Comment:
We should have default values defined as the users are not supposed to define all the parameters every time.

Reply:
Sure this makes more sense.

Reply:
fixed in [118fa23](https://github.com/theochem/procrustes/pull/195/commits/118fa237e6143020fa876793cfc3b17c4f780a46)

Reply:
@FanwangM I think we can't pass the Dictionary as a parameter since it is mutable and it will get passed by reference.

Reply:
In the current push, we have `options: Dict = None` as the default value to the parameter and the user doesn't need to pass anything.